src/features/saveToList/useSaveObjectToGroup.ts

```diff
@@ -0,0 +1,103 @@
+
+export const useSaveObjectToGroup = (node: LayoutNode<'List' | 'Checkboxes' | 'MultipleSelect'>) => {
+  const bindings = useNodeItem(node, (i) => i.dataModelBindings) as IDataModelBindingsForGroupCheckbox;
+  const checkedBinding = bindings.checked;
+  const checkedBindingSegments = checkedBinding?.field.split('.');
+  const checkedBindingKey = checkedBindingSegments?.pop();
+
+  const setLeafValue = FD.useSetLeafValue();
+  const { formData } = useDataModelBindings(bindings, DEFAULT_DEBOUNCE_TIMEOUT, 'raw');
+  const appendToList = FD.useAppendToList();
+  const removeFromList = FD.useRemoveIndexFromList();
+
+  const getObjectFromFormDataRow = (row: Row): Row | undefined =>
+    (formData?.group as Row[])?.find((selectedRow) =>
+      Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]),
+    );
+
+  const getIndexFromFormDataRow = (row: Row) =>
+    (formData?.group as Row[]).findIndex((selectedRow) => {
+      const { altinnRowId: _ } = selectedRow;
+      return Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]);
+    });
+
+  const isRowChecked = (row: Row): boolean => {
+    const formDataObject = getObjectFromFormDataRow(row);
+    if (checkedBindingKey) {
+      return !!formDataObject && !!formDataObject[checkedBindingKey];
+    }
+    return !!formDataObject;
+  };
+
+  const setList = (row: Row) => {
+    if (!bindings.group) {
+      return;
+    }
+    if (isRowChecked(row)) {
+      const index = getIndexFromFormDataRow(row);
+      if (checkedBinding && checkedBindingSegments) {
+        setLeafValue({
+          reference: {
+            ...bindings.checked,
+            field: `${checkedBindingSegments.join('.')}[${index}].${checkedBindingKey}`,
+          } as IDataModelReference,
+          newValue: false,
+        });
+      } else {
+        if (index >= 0) {
+          removeFromList({
+            reference: bindings.group,
+            index,
+          });
+        }
+      }
+    } else {
+      const formDataObject = getObjectFromFormDataRow(row);
+
+      if (formDataObject && checkedBindingKey && checkedBindingKey in formDataObject) {
+        const index = getIndexFromFormDataRow(row);
+        setLeafValue({
+          reference: {
+            ...bindings.checked,
+            field: `${checkedBindingSegments?.join('.')}[${index}].${checkedBindingKey}`,
+          } as IDataModelReference,
+          newValue: true,
+        });
+      } else {
+        const uuid = uuidv4();
+        const next: Row = { [ALTINN_ROW_ID]: uuid };
+        if (checkedBindingKey) {
+          next[checkedBindingKey] = true;
+        }
+        for (const binding of Object.keys(bindings)) {
+          if (binding === 'simpleBinding') {
+            const propertyName = bindings.simpleBinding.field.split('.').pop();
+            if (propertyName) {
+              next[propertyName] = row[propertyName];
+            }
```

Comment:
This is also seems to require that bindings are only one property deep, which shouldn't be a requirement.